compiler/src/test/scala/org/scalajs/nscplugin/test/JSInteropTest.scala

```diff
@@ -400,33 +623,125 @@ class JSInteropTest extends DirectTest with TestHelpers {
 
   }
 
+  @Test def noJSGlobalScopeExceptOnObjects: Unit = {
+    """
+    @js.native @JSGlobalScope
+    class A extends js.Any
+
+    @js.native @JSGlobalScope
+    trait B extends js.Any
+
+    object Container {
+      @js.native @JSGlobalScope
+      class C extends js.Any
+
+      @js.native @JSGlobalScope
+      trait D extends js.Any
+
+      @js.native @JSGlobalScope
+      val a: Int = js.native
+
+      @js.native @JSGlobalScope
+      def b: Int = js.native
+
+      @js.native @JSGlobalScope
+      def c(x: Int): Int = js.native
+    }
+    """ hasErrors
+    """
+      |newSource1.scala:5: error: Only native JS objects can have an @JSGlobalScope annotation.
+      |    @js.native @JSGlobalScope
+      |                ^
+      |newSource1.scala:8: error: Traits may not have an @JSGlobalScope annotation.
+      |    @js.native @JSGlobalScope
+      |                ^
+      |newSource1.scala:12: error: Only native JS objects can have an @JSGlobalScope annotation.
+      |      @js.native @JSGlobalScope
+      |                  ^
+      |newSource1.scala:15: error: Traits may not have an @JSGlobalScope annotation.
+      |      @js.native @JSGlobalScope
+      |                  ^
+      |newSource1.scala:18: error: Only native JS objects can have an @JSGlobalScope annotation.
+      |      @js.native @JSGlobalScope
+      |                  ^
+      |newSource1.scala:19: error: Native JS vals must have exactly one annotation among @JSGlobal, @JSImport and @JSGlobalScope.
```

Comment:
This error message is wrong :) It says `@JSGlobalScope` is OK. It's not :)

Reply:
Fixed. I also fixed it for classes, which were already wrong. Only `object`s can have `@JSGlobalScope`.